orchestrator/modules/agents/factory/agent_factory.py

```
@@ -865,13 +836,91 @@ async def _create_llm_manager(self, model_config: ModelConfiguration, agent_name
-        
+
         return LLMManager(config=llm_config)
+
+    async def _resolve_api_key(self, provider_name: str, agent_name: str = "", workspace_id=None) -> Optional[str]:
+        """
+        Resolve API key for a provider using the credential chain (PRD-54):
+        1. Credential store (development_<provider>)
+        2. BYOK user_api_keys table (workspace-scoped — MUST filter by workspace_id)
+        3. Environment variables
+        """
+        import os
```

Ruff (0.15.0)

[error] Redefinition of unused `os` from line 10: `os` redefined here

Remove definition: `os`

(F811)